metrics/network/network-metrics-iperf3.sh

```diff
@@ -0,0 +1,540 @@
+
+	metrics_json_init
+
+	# Start server
+	result=$(start_client "$image" "$client_command" "$client_extra_args")
+
+	# Save configuration
+	metrics_json_start_array
+
+	local result_line=$(echo "$result" | grep -m1 -E '\breceiver\b')
+	local -a results
+	read -a results <<< $result_line
+	local total_bandwidth=${results[6]}
+	local total_bandwidth_units=${results[7]}
+
+	local json="$(cat << EOF
+ 	{
+ 		"iperf3 network bandwidth": {
+			"Result" : $total_bandwidth,
+			"Units"  : "$total_bandwidth_units"
+		}
+	}
+EOF
+)"
+
+	metrics_json_add_array_element "$json"
+	metrics_json_end_array "Results"
+	metrics_json_save
+	clean_env
+}
+
+# Test jitter on single direction UDP
+function iperf3_jitter() {
+	local TEST_NAME="network iperf3 jitter"
+	# Start server
+	local server_address=$(start_server "$image" "$server_command" "$server_extra_args")
+
+	# Verify server IP address
+	if [ -z "$server_address" ];then
+		clean_env
+		die "server: ip address no found"
+	fi
+
+	metrics_json_init
+
+	# Start server
+	local client_command="$init_cmds && iperf3 -c ${server_address} -u -t ${transmit_timeout}"
+	result=$(start_client "$image" "$client_command" "$client_extra_args")
+
+	# Save configuration
+	metrics_json_start_array
+
+	local result_line=$(echo "$result" | grep -m2 -A2 -E '\bJitter\b' | tail -1)
+	local -a results
+	read -a results <<< $result_line
+	local total_jitter=${results[8]}
+	local total_jitter_units=${results[9]}
+
+	local json="$(cat << EOF
+	{
+		"iperf3 network jitter": {
+			"Result" : $total_jitter,
+			"Units"  : "$total_jitter_units"
+		}
+	}
+EOF
+)"
+
+	metrics_json_add_array_element "$json"
+	metrics_json_end_array "Results"
+	metrics_json_save
+	clean_env
+}
+
+# Run bi-directional TCP test, and extract results for both directions
+function iperf3_bidirectional_bandwidth_client_server() {
+	local TEST_NAME="network iperf3 bidirectional bandwidth"
+	# Start server
+	local server_address=$(start_server "$image" "$server_command" "$server_extra_args")
+
+	# Verify server IP address
+	if [ -z "$server_address" ];then
+		clean_env
+		die "server: ip address no found"
+	fi
+
+	metrics_json_init
+
+	# Start client
+	local client_command="$init_cmds && iperf3 -c ${server_address} -d -t ${transmit_timeout}"
+	result=$(start_client "$image" "$client_command" "$client_extra_args")
+
+	# Save configuration
+	metrics_json_start_array
+
+	local client_result=$(echo "$result" | grep -m1 -E '\breceiver\b')
+	local server_result=$(echo "$result" | grep -m1 -E '\bsender\b')
+	local -a client_results
+	read -a client_results <<< ${client_result}
+	read -a server_results <<< ${server_result}
+	local total_bidirectional_client_bandwidth=${client_results[6]}
+	local total_bidirectional_client_bandwidth_units=${client_results[7]}
+	local total_bidirectional_server_bandwidth=${server_results[6]}
+	local total_bidirectional_server_bandwidth_units=${server_results[7]}
+
+	local json="$(cat << EOF
+	{
+		"network iperf3 bidir bw client to server": {
```

Comment:
The long section name here is probably fine, but it might become tedious to parse out of the JSON, and most of that info is encoded in the TEST_NAME already. I'd suggest just naming those sections 'to server' and 'to client' maybe.